### PR TITLE
[INLONG-6598][Sort] Import small file compact for Apache Iceberg

### LINKED ISSUE #6598: [Improve][Sort] Import small file compact for Apache Iceberg
### Description

Import small file compact for iceberg

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by thesumery**
### Prepare a Pull Request
- Title Example: [INLONG-6598][Sort] Import small file compact for iceberg
- Fixes #6598 

### Motivation

Import a generic periodic iceberg action task at flink runtime.
1. Because flink is a stream task and do a periodic execution at checkpoint, so we can do some optimization item for iceberg(e.g small files compaction、expired manifest clean up)
2. dlc connector import small file compact, but import some repeated code, so this pr want to import this feature，so dlc connector can not import repeated code anymore.

### Modifications

Add an 'action-impl' plugin as an Iceberg Action provider

```
CREATE TABLE inlong_iceberg13  (
    id bigint,
    name string,
    PRIMARY KEY(id) NOT ENFORCED
) WITH (
    'connector'='iceberg-inlong',
    'catalog-name'='hive_prod',
    'catalog-database'='inlong',
    'catalog-table'='inlong_iceberg13',
    'uri'='thrift://localhost:9083',
    'warehouse'='hdfs://localhost:8020/hive/warehouse',
    'action-impl' = 'org.apache.inlong.sort.iceberg.FlinkActions$FlinkDefaultActions'  -- default action impl
);
```

So here we can instance 'org.apache.inlong.sort.iceberg.FlinkActions$FlinkDefaultActions'  as ActionProvider to provide, we can also define a new Class impelement FlinkActions which include our uniquet action.

### CODE REVIEW

File: inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/FlinkActions.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+package org.apache.inlong.sort.iceberg;
+
+import org.apache.iceberg.Table;
+import org.apache.iceberg.actions.ActionsProvider;
+import org.apache.iceberg.actions.RewriteDataFiles;
+
+import java.io.Serializable;
+import java.util.Map;
+
+public interface FlinkActions extends ActionsProvider, Serializable {
```
Comment (healchow) [inline inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/FlinkActions.java:29]: Add Java doc, please.

File: inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/FlinkConfigOptions.java [hunk 2]
```diff
@@ -22,6 +22,9 @@
 import org.apache.flink.configuration.ConfigOption;
 import org.apache.flink.configuration.ConfigOptions;
 
+/**
+ * Iceberg unique flink configuration
+ */
 public class FlinkConfigOptions {
 
     public static final ConfigOption<Boolean> TABLE_EXEC_ICEBERG_INFER_SOURCE_PARALLELISM =
```
Comment (EMsnap) [inline inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/FlinkConfigOptions.java:30]: TABLE_EXEC_ICEBERG_INFER_SOURCE_PARALLELISM
TABLE_EXEC_ICEBERG_INFER_SOURCE_PARALLELISM_MAX
they are not being used in current code, is it nessasary to keep them ?

File: inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/FlinkDynamicTableFactory.java [hunk 3]
```diff
@@ -64,37 +65,43 @@
...
- * IcebergTableSink}.We modify KafkaDynamicTableSink to support append-mode .
+ * IcebergTableSink}.We modify IcebergDynamicTableSink to support append-mode .
  * </p>
  */
 public class FlinkDynamicTableFactory implements DynamicTableSinkFactory, DynamicTableSourceFactory {
 
     static final String FACTORY_IDENTIFIER = "iceberg-inlong";
 
-    private static final ConfigOption<String> CATALOG_NAME =
+    public static final ConfigOption<String> CATALOG_NAME =
```
Comment (EMsnap) [inline inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/FlinkDynamicTableFactory.java:75]: why public this ?

File: inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/FlinkActions.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        private Map<String, String> actionProperties;
+
+        @Override
+        public void init(Map<String, String> actionProperties) {
+            this.actionProperties = actionProperties;
+        }
+
+        @Override
+        public RewriteDataFiles rewriteDataFiles(Table table) {
+            return FlinkActions.super.rewriteDataFiles(table);
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/FlinkActions.java:47]: Missing default implementation?

**Review by yunqingmoswu — COMMENTED** [review]
Can you add a detailed design description?

File: inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/FlinkConfigOptions.java (hunk 2 above)
Comment (thesumery) [reply inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/FlinkConfigOptions.java:30]: ok, remove this class

File: inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/FlinkActions.java (hunk 4 above)
Comment (thesumery) [reply inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/FlinkActions.java:47]: > Can you add a detailed design description?

ok，i will add a detail design here later in pr description.

**Comment by gong** [comment]
@thesumery please git rebase master, and exec mvn spotless:apply

**Review by yunqingmoswu — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Review by gong — APPROVED** [review]

**Pull request merged**